dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -48,18 +51,75 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
     winrt::event_token PushNotificationChannel::PushReceived(winrt::Windows::TypedEventHandler<winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel, winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler)
     {
-        return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
+        if (IsPackagedApp())
+        {
+            return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
+            {
+                auto strong = weak_self.get();
+                if (strong)
+                {
+                    handler(*strong, winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args));
+                };
+            });
+        }
+        else
         {
-            auto strong = weak_self.get();
-            if (strong)
+            if (!m_isRegisteredWithLRP)
             {
-                handler(*strong, winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args));
-            };
-        });
+                wil::com_ptr<INotificationsLongRunningPlatform> notificationsLongRunningPlatform{
+                    wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+                wchar_t processName[1024];
```

Comment:
Dangerous to assume that processName which includes the file path is max of 1024 characters.

Reply:
Looks like there is a MAX_PATH constant which equals 260. Will switch to this constant.

Reply:
Thats even worse. We can have file paths greater than 260 characters right? Then what?